services/apps/alcs/src/portal/notice-of-intent-submission-draft/notice-of-intent-submission-draft.service.spec.ts

```diff
@@ -0,0 +1,181 @@
+        new NoticeOfIntentSubmission({
+          owners: [],
+          parcels: [],
+        }),
+      )
+      .mockResolvedValueOnce(new NoticeOfIntentSubmission());
+
+    mockSubmissionRepo.save.mockResolvedValue(new NoticeOfIntentSubmission());
+    mockParcelService.update.mockResolvedValue([]);
+    const mockTransaction = jest.fn();
+    mockSubmissionRepo.manager.transaction = mockTransaction;
+    mockNoiSubmissionStatusService.getCopiedStatuses.mockResolvedValue([]);
+
+    const draft = await service.getOrCreateDraft('fileNumber', mockUser);
+
+    expect(mockSubmissionRepo.findOne).toHaveBeenCalledTimes(3);
+    expect(mockSubmissionRepo.save).toHaveBeenCalledTimes(1);
+    expect(
+      mockNoiSubmissionStatusService.getCopiedStatuses,
+    ).toHaveBeenCalledTimes(1);
+    expect(draft).toBeDefined();
+  });
+
+  it('should delete a draft, attached parcels, statuses and owners', async () => {
+    mockSubmissionRepo.findOne.mockResolvedValue(
+      new NoticeOfIntentSubmission({
+        owners: [],
+        parcels: [],
+      }),
+    );
+
+    mockSubmissionRepo.remove.mockResolvedValue(new NoticeOfIntentSubmission());
+    mockParcelService.deleteMany.mockResolvedValueOnce([]);
+    mockNoiSubmissionStatusService.removeStatuses.mockResolvedValue({} as any);
+
+    await service.deleteDraft('fileNumber', mockUser);
+
+    expect(mockSubmissionRepo.findOne).toHaveBeenCalledTimes(1);
+    expect(mockSubmissionRepo.remove).toHaveBeenCalledTimes(1);
+    expect(mockParcelService.deleteMany).toHaveBeenCalledTimes(1);
+    expect(mockNoiSubmissionStatusService.removeStatuses).toHaveBeenCalledTimes(
+      1,
+    );
+  });
+
+  it('should load two submissions and save one as not draft when publishing', async () => {
+    mockSubmissionRepo.findOne.mockResolvedValue(
+      new NoticeOfIntentSubmission({
+        uuid: 'fake',
+        owners: [],
+        parcels: [],
+      }),
+    );
+
+    mockSubmissionRepo.delete.mockResolvedValue({} as any);
+    mockSubmissionRepo.save.mockResolvedValue(new NoticeOfIntentSubmission());
+    mockParcelService.deleteMany.mockResolvedValueOnce([]);
+    mockGenerateSubmissionDocumentService.generateUpdate.mockResolvedValue();
+    mockNoiSubmissionStatusService.removeStatuses.mockResolvedValue({} as any);
+
+    await service.publish('fileNumber', new User());
+
+    expect(mockSubmissionRepo.findOne).toHaveBeenCalledTimes(2);
+    expect(mockSubmissionRepo.delete).toHaveBeenCalledTimes(1);
+    expect(mockSubmissionRepo.delete).toBeCalledWith({ uuid: 'fake' });
+    expect(mockParcelService.deleteMany).toHaveBeenCalledTimes(1);
+    expect(mockSubmissionRepo.save).toHaveBeenCalledTimes(1);
+    expect(mockNoiSubmissionStatusService.removeStatuses).toHaveBeenCalledTimes(
+      1,
+    );
+    expect(mockSubmissionRepo.save.mock.calls[0][0].isDraft).toEqual(false);
+    // expect(
+    //   mockGenerateSubmissionDocumentService.generateUpdate,
+    // ).toHaveBeenCalledTimes(1);
```

Comment:
do you need this?

Reply:
This will be uncommented when we add PDFs back in